File: gym_management/gym/customization/customer/customer.js
frappe.ui.form.on('Customer', {
    refresh: function(frm){
        toggale_subscription_field(frm);
		console.log("Customer JS Loaded");
        // frappe.msgprint("hello");
      
        frm.add_custom_button(__('Subscription' ), function() {
            frappe.model.open_mapped_doc({
                method : "gym_management.gym.customization.customer.customer.make_subscription",
                frm : frm
            
            });
        }, __("Create"));
    
    }
});    
// customer ke under jab subscriotion active ho tabhi show hoti ha
function toggale_subscription_field(frm){
    if (frm.doc.custom_active_subscription){
        frm.set_df_property("custom_active_subscription", "hidden", 0);
        frm.set_df_property("custom_join_date", "hidden", 0);
    }
    else{
        frm.set_df_property("custom_active_subscription", "hidden", 1);
        frm.set_df_property("custom_join_date", "hidden", 1);
    }
}


frappe.ui.form.on('Customer', {
    refresh: function (frm) {

        if (!frm.doc.name) return;

        frappe.call({
            method: "gym_management.gym.customization.customer.customer.has_paid_invoice",
            args: {
                customer: frm.doc.name
            },
            callback: function (r) {
                if (!r.message) return;
                frm.add_custom_button(__('Fitness Profile'), function () {

                    frappe.model.open_mapped_doc({
                        method: "gym_management.gym.customization.customer.customer.make_fitness_profile",
                        frm: frm
                    });

                },__("Create"));

            }
        });
    }
});
frappe.ui.form.on('Customer', {
    refresh(frm){
        if(frm.doc.custom_active_subscription){
            frm.add_custom_button(__("Employee"),  () => {
                open_employee_dialog(frm);
                console.log("::::::::::::::::::::::::")
            
            },__("Create"));
        }

    }
});

function open_employee_dialog(frm){
    let row = frm.doc.portal_users[0];
    let d = new frappe.ui.Dialog({
        title : "Create Employee",
        fields : [
            {
                label : "First Name",
                fieldname : "first_name",
                fieldtype : "Data",
                default: frm.doc.customer_name,
                reqd: 1
            },
            {
                label : "Gender",
                fieldname : "gender",
                fieldtype : "Link",
                default: frm.doc.gender,
                reqd: 1
                
            },
            {
                label : "Date of Birth",
                fieldname : "date_of_birth",
                fieldtype : "Date",
                default: frm.doc.custom_date_of_birth,
                reqd: 1
                
            },
            {
                label : "Date of Joining",
                fieldname : "date_of_joining",
                fieldtype : "Date",
                // default: frm.doc.custom_date_of_birth,
                reqd: 1
                
            },
            {
                label : "Status",
                fieldname : "status",
                fieldtype : "Select",
                options : ["Active","Inactive", "Suspended", "Left"],
                // default: "Active",
                reqd: 1
                    
            },
            {
                label : "User",
                fieldname : "user",
                fieldtype : "Link",
                default: row.user
                // reqd: 1

            }

        ],
        primary_action_label : "Create Employee",
        primary_action(values){ 
            frappe.call({
                method: "gym_management.gym.customization.customer.customer.make_employee",
                args : {
                    first_name : values.first_name,
                    gender : values.gender,
                    date_of_birth : values.date_of_birth,
                    date_of_joining: values.date_of_joining,
                    status : values.status,
                    user : values.user
                },
                
            });
            d.hide();
            
            
        }
    });
    d.show();
}







    

File: gym_management/gym/customization/customer/customer.py
import frappe
from frappe.model.mapper import get_mapped_doc
@frappe.whitelist()
def make_subscription(source_name, target_doc=None):
    def postprocess(source, target):
        target.party_type = "Customer",
        target.party = source.customer_name
        target.custom_email = source.email_id
        
        target.set("plans", [])
        if source.custom_interested_plan:
            row = target.append("plans", {})
            row.plan = source.custom_interested_plan
            row.qty = 1
    doc = get_mapped_doc(
        "Customer",
        source_name,
        {
            "Customer": {
                "doctype" : "Subscription",
                "field_map": {
                    "customer_primary_contact": "customer_primary_contact"
                },
            },
        },
        target_doc,
        postprocess=postprocess
    )
    # frappe.throw(str(doc.as_dict()))
    return doc

@frappe.whitelist()
def has_paid_invoice(customer):
    return frappe.db.exists(
        "Sales Invoice",
        {
            "customer": customer,
            "status": "Paid"
        }
    )

# fitness buttion  customer par 
@frappe.whitelist()
def make_fitness_profile(source_name, target_doc=None):
    doc = get_mapped_doc(
        "Customer",
        source_name,
        {
            "Customer": {
                "doctype": "Fitness Profile",
                "field_map": {
                    "customer_name": "customer",
                    "custom_active_subscription": "subscription",
                    "custom_join_date" : "subscription_start_date",
                    "custom_assigned_trainer" : "trainer"
                    
                }
            }
        },
        target_doc
    )
    return doc

import frappe
@frappe.whitelist()
def make_employee(first_name,gender,date_of_birth,date_of_joining,status, user):
    existing_employee = frappe.db.exists("Employee",{"user_id":user})
    if existing_employee:
        return
    else:
        doc = frappe.new_doc("Employee")
        doc.first_name = first_name
        doc.gender = gender
        doc.date_of_birth = date_of_birth
        doc.date_of_joining = date_of_joining
        # doc.status = status
        doc.user_id = user
        doc.insert()




    


